spec/features/user_signs_up_spec.rb

```diff
@@ -25,11 +28,84 @@
     expect(page).to have_content('Sign Out')
   end
 
-  scenario 'provide invalid registration information' do
+  scenario 'provide valid registration information - with avatar' do
+
+    visit new_user_registration_path
+
+    fill_in 'Email', with: 'john@example.com'
+    fill_in 'Password', with: 'password'
+    fill_in 'Password confirmation', with: 'password'
+    fill_in 'Username', with: 'John'
+    attach_file('user[avatar]', Rails.root + 'spec/fixtures/default_av.jpg')
+
+    click_button 'Sign up'
+
+    # within[id]
+    # expect(find('img')['src']).to have_content "default_av.jpg"
+    expect(page).to have_content('Welcome! You have signed up successfully.')
+    expect(page).to have_content('Sign Out')
+
+
+
+
+
+
+  end
+
+  scenario 'provide invalid registration information - field left empty' do
     visit new_user_registration_path
 
     click_button 'Sign up'
     expect(page).to have_content("can't be blank")
     expect(page).to_not have_content('Sign Out')
   end
+
+  scenario 'provide invalid registration information - email not unique' do
+
+    user = FactoryGirl.create(:user)
+
+    visit new_user_registration_path
+
+    fill_in 'Email', with: user.email
+    fill_in 'Password', with: 'password'
+    fill_in 'Password confirmation', with: 'password'
+    fill_in 'Username', with: 'John'
+
+    click_button 'Sign up'
+
+    expect(page).to have_content("Email has already been taken")
+
+  end
+
+  scenario 'provide invalid registration information - username not unique' do
+
+    user = FactoryGirl.create(:user)
+
+    visit new_user_registration_path
+
+    fill_in 'Email', with: 'john@example.com'
+    fill_in 'Password', with: 'password'
+    fill_in 'Password confirmation', with: 'password'
+    fill_in 'Username', with: user.username
+
+    click_button 'Sign up'
+
+    expect(page).to have_content('Username has already been taken')
+
+  end
+
+  scenario 'provide invalid registration information - password does not match confirmation' do
```

Comment:
Line is too long. [95/80]<br>Prefer double-quoted strings unless you need single quotes to avoid extra backslashes for escaping.